tcms/core/tests/test_middleware.py

```diff
@@ -0,0 +1,23 @@
+import os
+import unittest
+from django.test import TransactionTestCase
+from django.contrib.messages import get_messages
+
+
+@unittest.skipUnless(
+    os.getenv('TEST_CHECK_UNAPPLIED_MIGRATIONS_MIDDLEWARE'),
+    'CheckUnappliedMigrationsMiddleware testing not enabled')
+class TestCheckUnappliedMigrationsMiddleware(TransactionTestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.unapplied_migration_message = ('You have 41 unapplied migration(s)'
+                                           '. See <a href="https://kiwitcms.'
+                                           'readthedocs.io/en/latest/installin'
+                                           'g_docker.html#initial-configuratio'
+                                           'n-of-running-container">documentat'
+                                           'ion</a>')
+
+    def test_unapplied_migrations(self):
+        response = self.client.get('/')
+        messages = list(get_messages(response.wsgi_request))
+        self.assertEqual(str(messages[0]), self.unapplied_migration_message)
```

Comment:
response is either a response containing an HTML page or a redirect (see get(follow=True)). Which means you can use assertContains against the response to search for strings in the HTML text.

Reply:
Why not just use the purpose built `get_messages`?

Reply:
because we want to know the message is actually shown on the page the user will see and because this is an extra step which isn't needed in this case.